runtime/vm/ClassInitialization.cpp

```diff
@@ -391,6 +391,16 @@ classInitStateMachine(J9VMThread *currentThread, J9Class *clazz, J9ClassInitStat
 					clazz = VM_VMHelpers::currentClass(clazz);
 				}
 				VM_ObjectMonitor::exitObjectMonitor(currentThread, initializationLock);
+
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+				/*Initialize a Flattened Class Cache */
```

Comment:
change this comment to:

`/* Preparation is the earliest point where the defaultValue would needed. I.e pre-filling static fields */`

Reply:
also, this code needs to be moved up before the `if (((UDATA)currentThread | J9ClassInitUnverified)` line.

After the allocation, `clazz = VM_VMHelpers::currentClass(clazz);` must be done, since object allocations may lose vm access